src/backup-metadata/README.md

```diff
@@ -1,30 +1,59 @@
-# Installation Steps
+# backup-metadata-generator
+A utility that scrapes Cloud Foundry APIs to provide metadata about a CF installation.
+It is designed to be invoked as a [Velero hook](https://velero.io/docs/v1.5/backup-hooks/).
 
-## Prequisite
-- You have a kubernetes cluster
-- You have backup-metadata tarball downloaded (`gs://tas-cf-metadata-test-artifacts`)
+For more information, see [this doc](https://docs.google.com/document/d/1aR6_v0wTrSWpH9G2XqHcUTsdCNzP82ZsiWUxiN-4Zno/).
 
-## Installation instruction
-- Extract the tarball:
-`tar xvf backup-metadata.*.tgz && cd backup-metadata`
-- Create a folder `env-config`
-- Create a yaml file `values.yml` in `env-config` folder with contents:
+## Running Tests
+
+### Unit/Binary Tests
+```
+go test -v ./internal/... ./test/binary/...
+```
+
+### End-to-End (e2e) Integration Tests
+**Prerequisites:**
+
+1. A Kubernetes cluster with [Velero installed](https://velero.io/docs/v1.5/basic-install/#install-and-configure-the-server-components). See our [CI script](https://github.com/cloudfoundry/capi-ci/blob/master/ci/backup-metadata/install-velero.sh) for a GCP example.
+   For GCP you will also need to provide a service account with the following permissions:
+
+   ```
+   compute.disks.create
+   compute.disks.createSnapshot
+   compute.disks.get
+   compute.snapshots.create
+   compute.snapshots.delete
+   compute.snapshots.get
+   compute.snapshots.useReadOnly
+   storage.objects.create
+   storage.objects.get
+   storage.objects.delete
+   storage.objects.list
+   compute.zones.get
+   storage.buckets.get
+   ```
+
+   For other IaaSes refer to the Velero docs.
+2. The [`velero` CLI installed locally](https://velero.io/docs/v1.5/basic-install/#install-the-cli)
+3. `kubectl` targeted to this environment
+4. GNU grep as `grep` on your PATH (`brew install grep`)
```

Comment:
I created a separate chore to remove this requirement:
https://www.pivotaltracker.com/story/show/175563847